sqld/src/database/libsql.rs

```diff
@@ -344,9 +345,37 @@ fn eval_cond(cond: &Cond, results: &[Option<QueryResult>]) -> Result<bool> {
     })
 }
 
+fn check_auth(auth: Authenticated, pgm: &Program) -> Result<()> {
+    for step in pgm.steps() {
+        let query = &step.query;
+        match (query.stmt.kind, &auth) {
+            (_, Authenticated::Anonymous) => {
+                return Err(Error::NotAuthorized(
+                    "anonymous access not allowed".to_string(),
+                ));
+            }
+            (StmtKind::Read, Authenticated::Authorized(_)) => (),
+            (StmtKind::TxnBegin, _) | (StmtKind::TxnEnd, _) => (),
+            (_, Authenticated::Authorized(Authorized::FullAccess)) => (),
+            _ => {
+                return Err(Error::NotAuthorized(format!(
+                    "Current session is not authorized to run: {}",
+                    query.stmt.stmt
+                )));
+            }
+        }
+    }
+    Ok(())
+}
```

Comment:
for a followup PR: this is too restrictive, we should only deny steps that will be executed.